### ISSUE TITLE
using --password should provide a prompt

### ISSUE THREAD

**Issue by kmf** [inspec/inspec#1997]
### Description

using --password should provide a prompt for a password if none is specified.

### Ouput
```
/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/inspec-1.30.0/lib/inspec/base_cli.rb:113:in `block in opts': Please provide a value for --password. For example: --password=hello. (ArgumentError)
```

### InSpec and Platform Version
- inspec 1.30
- Ubuntu 16.04.2

##### Sidenote: I could have sworn it did work like this .... maybe I'm getting old.

**Comment by adamleff** [comment]
@kmf you're not getting old... well, we're all getting old, but that's not the reason for your perception of this issue. 🙂 

We never supported InSpec itself prompting for a password. What was actually happening was when you provided `--password` without passing a value, InSpec would pass an empty password string to the SSH library which would try to log in with an empty password. The SSH library itself was prompting for the password, but not until after it tried to do a login and the target rejected the login.

In #1918, we enforced passing an actual value because we received reports of people accidentally logging themselves out of their remote systems because of too many failed login attempts!

I personally prefer not prompting by default if a user passes `--password` without a value as I fear this could cause unattended uses to get hung waiting for input instead of raising an exception like we do now. But I think this could be optional behavior... i.e. adding a flag `--prompt-for-password` and `--prompt-for-sudo-password` so that the password isn't shown in cleartext on the command line. Would that meet your needs? Obviously we'd rather you use SSH keys instead of passwords.

@arlimus what do you think of us implementing the `--prompt-for-foo` flags above to allow for interactive password input?

**Comment by kmf** [comment]
I was busy doing some training for some colleagues on Inspec, so I made them look away a few times, I would like a option eg. 
```
--ask-pass
```

**Comment by adamleff** [comment]
Yeah, I agree we could make this better.

Until then, if you find yourself in the same situation, set your password to a variable in your shell and then your users won't see it. Yeah, I know, it's ugly and you shouldn't need to do it, but I at least want to tell you what I would do in your situation if I couldn't use SSH keys:

```
MYPASSWORD="sooperseekret"
<clear screen>
inspec exec myprofile --target ssh://me@hostname --password ${MYPASSWORD}
```

**Comment by adamleff** [comment]
Duplicate of #1902 

@kmf I'm going to close this issue in favor of #1902 in which we hope to provide a CLI flag to prompt the user for the password. We originally had to cancel those plans because of some limitations in Thor but we believe we have a reasonable workaround that we can implement in an effort to provide this functionality. Please watch that issue for further details. Thanks!

**Comment by kmf** [comment]
@adamleff it's cool.

**Comment by RubyTuesdayDONO** [comment]
> set your password to a variable in your shell and then your users won't see it.

sorry for resurrecting an ancient issue, but respectfully, the suggested work-around should be surrounded by GIANT WARNING FLAGS not to ever use it, because it will leave the user's password both in their shell history file, in `ps` (running process list), and in the system's audit logs (if `execve` is logged). 

security of confidential data is much bigger than someone reading over your shoulders.

**Comment by aaronlippold** [comment]
Oh, I agree that this is an issue. When we talked about it in the last
community meaning many moons ago the general decision was that Thor was not
quite able to meet this requirement and that migration to a new CLI system
would be the best solution to formuly to solve this problem. I'm not sure
if that's still the current agreement but I think it's good that we started
the conversation again.

**Comment by aaronlippold** [comment]
Also, another workaround could be setting up credentials in the inspect
config file but this would not address all the issues of confidentiality
but it would address part of them.. This is in no way to say that we have a
complete solution around this challenge and I would be happy to be part of
the conversation to get us to a full solution.
